### DISCUSSION TITLE
Automated release process?

### DISCUSSION THREAD

**Discussion by luwes** [muxinc/media-chrome#239] [Polls]
An automated NPM publish by a Github action when merging to the `main` branch with a `prerelease` version (+ commit SHA) on a specific NPM dist tag. Makes sense to use a similar version format Lerna is using for this since we use this in Mux elements. See link below.

The added commit hash (SHA) is useful because CI/CD runs possibly in parallel, multiple people w/ several feature branches -> avoid version collisions.

Handy resources:
- https://github.com/lerna/lerna/tree/main/commands/publish#--canary
- https://docs.npmjs.com/cli/v8/commands/npm-dist-tag
- https://liquidsoftware.com/blog/the-7-deadly-sins-of-versioning-part-2/

### Question 2: What should the NPM dist tag (release channels) be called?
- next
- canary
- beta
- stable
- dev
- other

Please answer with comment

### Question 3: Should we also do this for Pull Requests? 
- yes
- no

If yes, please answer yes with answers for question 1 (prerelease tag?) & 2 (dist tag?)

**Comment by luwes** [comment]
- Question 1: I chose `alpha` e.g. 1.1.0-alpha.0+81e3b443 
  `beta` or `rc` feels more mature than what's merged in the main branch each time. keep `canary` for PR's.
- Question 2: I chose `dev` channel. keep `next` for beta's and RC's. keep `canary` for PR's.
- Question 3: not yet I think, until there is a need for it
- Question 4: yes, provide a GH action manual trigger to create a normal NPM release

**Comment by dylanjha** [comment]
@luwes 

We discussed this on a call. What we're landing on is:

- All commits on `main` get a published `canary` version: They will be on the `@canary` NPM release channel and be versioned like: `1.1.0-canary+81e3b443`
- GH action to manually trigger a normal NPM release

In this set up we will have 2 NPM release channels:

- `latest` (default, normal production release channel)
- `canary` (every commit on main)

And we will also want a `next` release channel:

- `next` (the next release, that we consider stable and want people to try before making it `latest`)
